internal/utils/aggregator_test.go

```diff
@@ -0,0 +1,230 @@
+	usage_data = []UsageData{
+		{
+			"2023-02-01 00:00:00 +0000 UTC", "2023-03-01 00:00:00 +0000 UTC", "2023-04-02 00:00:01 +0000 UTC", "2023-04-02 00:15:00 +0000 UTC",
+			"Yuptoo-service", "Yuptoo-app-standalone-1", "Yuptoo-app", "ReplicaSet", "<none>", "deployment", "Yuptoo-prod",
+			"quay.io/cloudservices/yuptoo", "ip-10-0-176-227.us-east-2.compute.internal", "i-0dfbb3fa4d0e8fc94",
+			"1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1",
+		},
+	}
+	df = dataframe.LoadStructs(usage_data)
+	result = Aggregate_data(df)
+
+	if result.Nrow() == 0 {
+		t.Errorf("Aggregator function hasn't clean the data properly. (case - all columns present) Expected - 1 but got %v", result.Nrow())
+	}
+}
+
+func TestAggregateAddingNewColumn(t *testing.T) {
+	usage_data := []UsageData{
+		{
+			"2023-02-01 00:00:00 +0000 UTC", "2023-03-01 00:00:00 +0000 UTC", "2023-04-02 00:00:01 +0000 UTC", "2023-04-02 00:15:00 +0000 UTC",
+			"Yuptoo-service", "Yuptoo-app-standalone-1", "Yuptoo-app", "ReplicaSet", "<none>", "deployment", "Yuptoo-prod",
+			"quay.io/cloudservices/yuptoo", "ip-10-0-176-227.us-east-2.compute.internal", "i-0dfbb3fa4d0e8fc94",
+			"1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1",
+		},
+	}
+	df := dataframe.LoadStructs(usage_data)
+	result := Aggregate_data(df)
+
+	if result.Nrow() != 1 {
+		t.Errorf("Expected number of row - 1 but got %v", result.Nrow())
+	} else {
+		data := result.Maps()[0]
+		if data["k8s_object_type"] != "replicaset" || data["k8s_object_name"] != "Yuptoo-app" {
+			t.Errorf("Expected k8s_object_type = replicaset and k8s_object_name = Yuptoo-app but got k8s_object_type = %s and k8s_object_name = %s", data["k8s_object_type"], data["k8s_object_name"])
+		}
+	}
+
+}
+
+func TestAggregateData(t *testing.T) {
+	usage_data := []UsageData{
+		{
+			"2023-02-01 00:00:00 +0000 UTC", "2023-03-01 00:00:00 +0000 UTC", "2023-04-02 00:00:01 +0000 UTC", "2023-04-02 00:15:00 +0000 UTC",
+			"Yuptoo-service", "Yuptoo-app-standalone-1", "Yuptoo-app", "ReplicaSet", "<none>", "deployment", "Yuptoo-prod", "quay.io/cloudservices/yuptoo",
+			"ip-10-0-176-227.us-east-2.compute.internal", "i-0dfbb3fa4d0e8fc94", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1",
+		},
+		{
+			"2023-02-01 00:00:00 +0000 UTC", "2023-03-01 00:00:00 +0000 UTC", "2023-04-02 00:00:01 +0000 UTC", "2023-04-02 00:15:00 +0000 UTC",
+			"Yuptoo-service", "Yuptoo-app-standalone-2", "Yuptoo-app", "ReplicaSet", "<none>", "deployment", "Yuptoo-prod", "quay.io/cloudservices/yuptoo",
+			"ip-10-0-176-227.us-east-2.compute.internal", "i-0dfbb3fa4d0e8fc94", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2", "2",
+		},
+	}
+
+	df := dataframe.LoadStructs(usage_data)
+	result := Aggregate_data(df)
+	if result.Nrow() != 1 {
```

Comment:
@patilsuraj767 I didn't quite get this part, not sure why expected row count is 1.

Reply:
As an input `usage_data` we are feeding two records of two pods but those to pods are replica of same deployment. In this case it is expected that aggregate function combine those two records into one, that is the reason we are only expecting 1 row in output.